### PR TITLE
docs: refactor to improve flow + general review and update

### PR DESCRIPTION
**Pull request by edvald**
See it live at https://docs.garden.io/v/docs%2Fexperimental/

TODO
- [x] Review Using Garden section
- [x] Document environments and namespaces
- [x] Split bulk of Configuration Files guide into Projects + Modules
- [x] Add section on module types
- [x] Validate and update all links
- [x] Add images in How Garden Works + README
- [x] Create custom landing page for docs site
- [x] Documentation overview section in README

### CODE REVIEW

File: docs/guides/development-workflows.md [hunk 1]
```diff
@@ -1,10 +1,8 @@
 # Development Workflows
```
[the remark this answers is not in the document]
Comment (edvald) [reply docs/guides/development-workflows.md:1]: Agreed 👍

**Comment by eysi09** [comment]
This looks great (at a glance)! I skimmed the structure but I won't have type for a proper review until the weekend.

File: README.md [hunk 2]
```diff
@@ -2,196 +2,147 @@
...
+- **Test orchestration:** Garden has first-class support for testing, and allows you to easily define tests that run in-cluster, alongside your services.
+- **Result caching:** Built and test results are shared between users when using the same cluster. Teams can easily share build and test results, and your CI can use the same build caches as the team.
 
-### macOS
+| Resources | What is it? |
+| --------- | ----------- |
+| [Garden Website](https://garden.io) | The official Garden website, including information about the supported enterprise edition of the product |
+| [Garden Documentation / Getting Started](https://docs.garden.io) | Documentation for all editions of Garden, and the best place to get started |
+| [Blog](https://medium.com/garden-io) | Our company blog, where we share product updates, how-to guides, and other resources |
+| [Community Slack](https://chat.garden.io) | The best place to ask questions as a user of open source Garden Core |
```
Comment (thsig) [inline README.md:18]: How about replacing "open source Garden core" with "Garden" here?

Comment (edvald) [reply README.md:18]: Sure 👍

File: docs/basics/how-garden-works.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Garden is pluggable by design and supports a variety of [providers](../reference/providers/README.md) and [module types](../reference/module-types/README.md), which you can choose based on preference and to suit your existing set-up.
+
+When you run Garden with a shared Kubernetes cluster, Garden has its own namespace with a small set of services, including:
+
+- **Image cache and test cache:** Every time an image is built for a service, Garden saves it a cache. If a different developer wants to deploy the same version of an image, the image doesn’t have to be rebuilt–instead, Garden will check if the image exists in the Docker registry and skip the build step if it does. Test results are also saved in a test cache and can be shared across developers. These caches can reduce build and test time significantly, often by 80% or more.
+- **Build service:** Developers and CI processes don’t need to run Docker and Kubernetes on their machine (unless they want to). The build service makes it possible for all of this to be handled in the cluster.
+
+![Cluster Diagram](../cluster-diagram.png "Garden + Kubernetes Cluster Diagram")
```
[the remark this answers is not in the document]
Comment (edvald) [reply docs/basics/how-garden-works.md:21]: Moving it could make sense, yes.

**Comment by edvald** [comment]
> On a more general note, I'm wondering if we're missing more introductory content on the What and Who. I.e. who is Garden for and what would they use it for.

That's a good idea. Not 100% sure how we'd define that though, just yet.

**Pull request merged**